Research/ResearchUI/iOS/Step View Controllers/RSDActiveStepViewController.swift

```diff
@@ -91,7 +95,40 @@ open class RSDActiveStepViewController: RSDFullscreenImageStepViewController {
     /// The `countdownFormatter` is used to format the string derived from this time interval.
     open override var countdown: Int {
         didSet {
+            updateCountdownLabels()
+        }
+    }
+    
+    /// Update the countdown labels
+    open func updateCountdownLabels() {
+        guard let stepDuration = self.activeStep?.duration
+            else {
+                unitLabel?.text = nil
+                countdownLabel?.text = nil
+                doneLabel?.isHidden = true
+                return
+        }
+        
+        let countdown = (self.clock == nil) ? Int(stepDuration) : self.countdown
+        if countdown == 0 {
+            unitLabel?.text = nil
+            countdownLabel?.text = nil
+            doneLabel?.isHidden = false
+        }
+        else {
+            doneLabel?.isHidden = true
             countdownLabel?.text = countdownFormatter.string(from: TimeInterval(countdown))
+            if self.allowedCountdownUnits == [.second] {
+                switch countdown {
+                case 1:
```

Comment:
keep in mind we'd need to update singular/plural selection code like this if we ever need to support a Slavic language, as they can have multiple different plural forms depending on the last digit of the number...

Reply:
Yes. *sigh* I cannot figure out the magic for getting `.stringsdict` files to work for other than the "Localizable.stringsdict" in the main bundle. Decided that it was a task for another day.

Really, this *should* use the built in function for time units, but *that* doesn't work for time. (works for other types of units).